BynderExtension/Extension/AssetLoader.cs

```diff
@@ -1,53 +1,133 @@
+
+        public AssetLoader()
+        {
+            _lazyConnectorState = new Lazy<ConnectorState>(() =>
+            {
+                var connectorStates = Context.ExtensionManager.UtilityService.GetAllConnectorStatesForConnector(Context.ExtensionId);
+                if (connectorStates.Any())
+                {
+                    return connectorStates.Select(state => (state.Modified, state: state)).Max().state;
+                }
+
+                var newConnectorState = new ConnectorState
+                {
+                    ConnectorId = Context.ExtensionId,
+                    Data = null
+                };
+                return Context.ExtensionManager.UtilityService.AddConnectorState(newConnectorState);
+            });
+
+            _lazyLastRunTime = new Lazy<DateTime?>(() =>
+            {
+                if (string.IsNullOrEmpty(ConnectorState.Data))
+                {
+                    return null;
+                }
+
+                try
+                {
+                    return JsonConvert.DeserializeObject<DateTime?>(ConnectorState.Data);
+                }
+                catch (JsonException jsonException)
+                {
+                    Context.Log(LogLevel.Error,
+                        $"Failed to deserialize connector state data as DateTime?. Data: {ConnectorState.Data}",
+                        jsonException);
+                }
+
+                return null;
+            });
+        }
+
         /// <summary>
         /// Get a list of all assetIds from Bynder using the configured filter Query
         /// which will be executed against api/v4/media/?-----
         /// for each found asset, process it using the worker implementation as if it would have been triggered by a notificaton message
         /// </summary>
         public void Execute(bool force)
         {
-            // only run manually
-            if (!force) return;
-
             try
             {
                 Context.Logger.Log(LogLevel.Information, "Start loading assets");
 
                 var worker = Container.GetInstance<AssetUpdatedWorker>();
                 var bynderClient = Container.GetInstance<BynderClient>();
+                var lastRunTime = FullSync() ? null : LastRunTime;
+                var startTime = DateTime.Now;
 
                 // get all assets ids
                 // note: this is a paged result set, call next page until reaching end.
                 var counter = 0;
                 var assetCollection = bynderClient.GetAssetCollection(Context.Settings[Config.Settings.InitialAssetLoadUrlQuery]);
                 Context.Logger.Log(LogLevel.Information, $"Start processing {assetCollection.GetTotal()} assets.");
 
-                assetCollection.Media.ForEach(a => worker.Execute(a.Id));
-                counter += assetCollection.Media.Count;
-                while (!assetCollection.IsLastPage())
+                while(true)
```

Comment:
Perhaps do while (!asserCollection.lastpage) could be used. Doesnt really matter... 😛